packages/@sanity/cli/src/actions/typegen/__tests__/generateAction.test.ts

```diff
@@ -243,6 +243,10 @@ test(generateAction.name, async () => {
 
     export declare const internalGroqTypeReferenceTo: unique symbol;
 
+    export type ArrayMember<T> = T & {
+      _key: string;
+    };
```

Comment:
Where does this come from? 🤔 
I think a type like this should be defined in `@sanity/codegen` or `@sanity/types` instead of being repeated here?